Gems/Atom/Feature/Common/Code/Source/Shadows/FullscreenShadowPass.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Shadows/FullscreenShadowPass.h>
+
+#include <Atom/RHI/Factory.h>
+#include <Atom/RHI/PipelineState.h>
+#include <Atom/RPI.Public/Pass/PassUtils.h>
+#include <Atom/RPI.Public/RenderPipeline.h>
+#include <Atom/RPI.Public/RPIUtils.h>
+#include <Atom/RHI/RHISystemInterface.h>
+#include <Atom/RPI.Reflect/Pass/PassTemplate.h>
+#include <Atom/RPI.Public/View.h>
+
+namespace AZ
+{
+    namespace Render
+    {
+        RPI::Ptr<FullscreenShadowPass> FullscreenShadowPass::Create(const RPI::PassDescriptor& descriptor)
+        {
+            RPI::Ptr<FullscreenShadowPass> pass = aznew FullscreenShadowPass(descriptor);
+            return pass;
+        }
+
+        FullscreenShadowPass::FullscreenShadowPass(const RPI::PassDescriptor& descriptor)
+            : RPI::FullscreenTrianglePass(descriptor)
+            , m_outputName("Output")
+            , m_depthInputName("Depth")
+        {
+        }
+
+        void FullscreenShadowPass::CompileResources(const RHI::FrameGraphCompileContext& context)
+        {
+            SetConstantData();
+            FullscreenTrianglePass::CompileResources(context);
+        }
+
+        RPI::PassAttachmentBinding FullscreenShadowPass::GetPassAttachmentBinding(AZ::Name name)
+        {
+            for (unsigned int i = 0; i < GetInputCount(); ++i)
+            {
+                auto binding = GetInputBinding(i);
+                if (binding.m_name == name)
+                    return binding;
+            }
+
+            for (unsigned int i = 0; i < GetInputOutputCount(); ++i)
+            {
+                auto binding = GetInputOutputBinding(i);
+                if (binding.m_name == name)
+                    return binding;
+            }
+
+            for (unsigned int i = 0; i < GetOutputCount(); ++i)
+            {
+                auto binding = GetOutputBinding(i);
+                if (binding.m_name == name)
+                    return binding;
+            }
+            return {};
+        }
```

Comment:
@antonmic Seems like this could be a generic pass system utility.

Reply:
It is, see Pass::FindAttachmentBinding